test/widgets/test_redshift.py

```diff
@@ -0,0 +1,229 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from types import ModuleType
+
+import pytest
+
+from libqtile.config import Bar, Screen
+from libqtile.confreader import Config
+from libqtile.widget import redshift
+
+
+class MockRedshiftDriver(ModuleType):
+    """
+    Helper class as redshift is not available in
+    the test environment, so without this mock the tests will crash
+    """
+
+    def enable(self, temperature, gamma: redshift.GammaGroup, brightness):
+        pass
+
+    def reset(self):
+        pass
+
+
+def get_mock_rs_driver(_):
+    return MockRedshiftDriver("redshift_driver")
+
+
+@pytest.fixture(scope="function")
+def patched_redshift(monkeypatch):
+    class PatchedRedshift(redshift.Redshift):
+        def __init__(self, **config):
+            monkeypatch.setattr(
+                "libqtile.widget.redshift.Redshift._get_rs_driver", get_mock_rs_driver
+            )
+            redshift.Redshift.__init__(self, **config)
+            self.name = "redshift"
+
+    yield PatchedRedshift
+
+
+@pytest.fixture(scope="function")
+def redshift_manager(manager_nospawn, request, patched_redshift):
+    class GroupConfig(Config):
+        screens = [
+            Screen(
+                top=Bar(
+                    [
+                        patched_redshift(
+                            update_interval=10,
+                            **getattr(request, "param", dict()),
+                        )
+                    ],
+                    30,
+                )
+            )
+        ]
+
+    manager_nospawn.start(GroupConfig)
+
+    yield manager_nospawn
+
+
+def config(**kwargs):
+    return pytest.mark.parametrize("redshift_manager", [kwargs], indirect=True)
+
+
+def test_defaults(redshift_manager):
+    widget = redshift_manager.c.widget["redshift"]
+
+    def text():
+        return widget.info()["text"]
+
+    def click():
+        redshift_manager.c.bar["top"].fake_button_press(0, 0, 1)
+
+    disabled_txt = "󱠃"
+    enabled_txt = "󰛨"
+    assert text() == disabled_txt
+
+    scroll_vals = [
+        "Brightness: 1.0",
+        "Temperature: 1700",
+        "Gamma: 1.0:1.0:1.0",
+        disabled_txt,
+    ]
+
+    for _, val in enumerate(scroll_vals):
+        widget.scroll_up()
+        assert text() == val
+
+    # move enabled_txt to the first index
+    scroll_vals.remove(disabled_txt)
+    scroll_vals = [disabled_txt] + scroll_vals
+
+    for _, val in enumerate(reversed(scroll_vals)):
+        widget.scroll_down()
+        assert text() == val
+
+    click()
+    assert text() == enabled_txt
+
+    click()
+    assert text() == disabled_txt
+
+
+@pytest.mark.parametrize(
+    "redshift_manager",
+    [
+        {
+            "disabled_txt": "Redshift disabled",
+            "enabled_txt": "Redshift enabled",
+        }
+    ],
+    indirect=True,
+)
```

Comment:
```suggestion
@config(disabled_txt="Redshift disabled", enabled_txt="Redshift enabled")
```
The `config` decorator simplifies the code here.